.github/workflows/unit-tests.yml

```diff
@@ -24,5 +24,6 @@ jobs:
       - name: Upload coverage to codecov.io
         uses: codecov/codecov-action@v3
         with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
they say for public repos no token is needed. Also, if we did this ^, then PRs won't be able to upload coverage because secrets are only passed for runs on the primary repo, not on forks.

Reply:
Even if jaegertracing and I don’t have a secret Token in our settings but with the token field defined the step is passing while without the other dependabot pr are failing , why is that happening then ?

Reply:
The problem is that those PRs are failing sporadically, it's not consistent on the presence of the token. It seems to be related to rate limiting on Codecov side (see https://github.com/codecov/codecov-action/issues/837)